CONTRIBUTING.md

```diff
@@ -76,3 +76,119 @@ issues.
 Although not a formal channel, you can also find a number of core developers on
 the #php.pecl channel on [EFnet](http://www.efnet.org/). Similarly, many
 documentation writers can be found on #php.doc.
+
+## PHP source code directory structure
+
+PHP source code also includes several files generated during development and
+several parts where maintenance is happening upstream in their respective
+locations.
+
+```bash
+<php-src>/
+ ├─ .git/                           # Git configuration and source directory
+ └─ TSRM/                           # Thread Safe Resource Manager
+    └─ m4/                          # https://github.com/autoconf-archive/autoconf-archive
+       └─ ax_func_which_gethostbyname_r.m4
+    └─ ...
+ └─ Zend/                           # Zend Engine
+    ├─ zend_language_scanner.c      # Generated by re2c 1.0.3
+    ├─ zend_language_scanner_defs.h # Generated by re2c 1.0.3
+    ├─ zend_ini_scanner.c           # Generated by re2c 1.0.3
+    ├─ zend_ini_scanner_defs.h      # Generated by re2c 1.0.3
+    ├─ zend_vm_execute.h            # Generated by `Zend/zend_vm_gen.php`
+    ├─ zend_vm_opcodes.h            # Generated by `Zend/zend_vm_gen.php`
+    ├─ zend_vm_opcodes.c            # Generated by `Zend/zend_vm_gen.php`
+    └─ ...
+ ├─ appveyor/                       # Appveyor CI service files
+ └─ build/                          # *nix build system files
+    ├─ ax_check_compile_flag.m4     # https://github.com/autoconf-archive/autoconf-archive
+    ├─ ax_gcc_func_attribute.m4     # https://github.com/autoconf-archive/autoconf-archive
+    ├─ libtool.m4                   # https://git.savannah.gnu.org/cgit/libtool.git
+    ├─ shtool                       # https://www.gnu.org/software/shtool/
+    └─ ...
+ └─ ext/                            # PHP core extensions
+    └─ bcmath/
+       ├─ libbcmath/                # Forked and maintained in php-src
+       └─ ...
+    └─ curl/
+       ├─ sync-constants.php        # The curl symbols checker
+       └─ ...
+    └─ date/
+       └─ lib/                      # Bundled datetime library https://github.com/derickr/timelib
+          ├─ parse_date.c           # Generated by re2c 0.15.3
+          ├─ parse_iso_intervals.c  # Generated by re2c 0.15.3
+          └─ ...
+       └─ ...
+    └─ fileinfo/
+       ├─ libmagic/                 # Bundled libmagic https://github.com/file/file
```

Comment:
Our libmagic is heavily patched, see [libmagic.patch](https://github.com/php/php-src/blob/master/ext/fileinfo/libmagic.patch) and [magicdata.patch](https://github.com/php/php-src/blob/master/ext/fileinfo/magicdata.patch).

Reply:
Updated with `Modified libmagic https://github.com/file/file` and mentioning patches. Thanks.